doc/tutorials/content/gpu_octree.rst

```diff
@@ -0,0 +1,235 @@
+
+.. literalinclude:: sources/gpu_octree/gpu_octree.cpp
+    :language: cpp
+    :lines: 22-45
+
+
+Building an octree
+------------------
+
+Next we create an octree instance and build the octree. In order to do this, we must first create an instance of `pcl::gpu::Octree::PointCloud` using the points in our point cloud. In order to do this, we must first upload those points to the GPU. Here "upload" refers to copying data from host memory to GPU memory (Conversely, "download" refers to copying data from GPU memory to host memory). We will also upload our queries for later use.  For this, we must create a `pcl::gpu::Octree::Queries` object and upload query points to this object.
+
+Note that unlike in the CPU octree implementation, the user cannot set the resolution of the octree. Instead, the depth of the octree will be automatically determined depending on the point distribution of the cloud (However, the maximum depth is capped at 10).
+
+.. literalinclude:: sources/gpu_octree/gpu_octree.cpp
+    :language: cpp
+    :lines: 47-58
+
+After building the octree, we we can move on to the different search queries.
+
+K Nearest Neighbors search
+---------------------------
+
+The K nearest neighbor algorithm is designed to identify a given (k) number of nearest neighbors to a query point. However, the GPU KNN search algorithm is currently restricted to k=1 only.
+
+
+.. note::
+
+    The GPU K nearest neighbor algorithm differs from it's CPU counterpart in implementation. Specifically, the CPU algorithm will only find approximate nearest neighbors, where the user can set the upper bound of the error allowed. In contrast, the GPU algorithm finds the absolute nearest neighbor.
+
+In order to perform K nearest search, we must prepare output buffers on the device to store the results of the search queries. Specifically, we need two separate buffers to store the resulting nearest neighbor indices and the corresponding squared distances between the search point and nearest neighbors. Note that these buffers will be automatically resized to `k * number of queries` size, in order to hold all the results.
+
+Now, we can perform the K nearest neighbors search on GPU. After performing the search we must download the results from GPU memory into host memory using two new vectors for indices and square distances.
+
+
+.. literalinclude:: sources/gpu_octree/gpu_octree.cpp
+    :language: cpp
+    :lines: 62-98
+
+Radius Search
+-------------
+
+Radius search returns neighbors within a given radius to the query point. The user must specify a maximum limit of points to return per query. There are implementations radius search available which either return / do not return the corresponding squared distances to the query point. Note that the methods which do not return squared distances are more efficient than those that do. In usage, this works very similar to the K nearest neighbor search.
```

Comment:
> methods which do not return squared distances are more efficient than those that do

I would say marginally more efficient, but I would like some results validating this.

Reply:
Yes, it's not a huge difference, but it is noticeable. These are the times it took for the two tests.

With distance returning
```shell
6: [       OK ] PCL_OctreeGPU.batchRadiusSearch (481 ms)
```

without distance returning
```shell
6: [       OK ] PCL_OctreeGPU.batchRadiusSearch (395 ms)
```

Reply:
Those number are a good hint  but that's not proper benchmarking. https://github.com/google/benchmark

Reply:
Does PCL's `ScopeTime` provide equivalent functionality as well, or is this much better?

Reply:
Performance based decision should be made with proper benchmarking. Example: 
https://github.com/PointCloudLibrary/pcl/pull/3890#discussion_r438477954